core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -144,9 +147,19 @@ private Schema initializeOutputSchema() {
    * @param inputSchema Validates whether the keyfield, fieldname, and fieldvalue entered by the user is of type
    *                    String or Nullable String and present in the input schema or not.
    * @param collector FailureCollector used to log all errors and return them to the user.
-   * @return true
+   * @returns true if input fields have been validated.
    */
-  private void validateInputFields(Schema inputSchema, FailureCollector collector) {
+  private boolean validateInputFields(Schema inputSchema, FailureCollector collector) {
+
+    // Bail out if input schema or field values are unknown.
+    if (inputSchema == null ||
```

Comment:
this is not what I mean, if one field is macro enabled, the other logic should still be validated. You should incorporate the this logic in the validation